fix: rewind pipe issues

shaik-zeeshan opened the pull request:
related: #1376 , #1474
/claim #1376

fixed skipping frames and showing date and time

#1376  this issues was happening when there were more than 1000+ frames element in DOM which was slowing down browsing timeline experience

before:

https://github.com/user-attachments/assets/72305926-1168-40c1-95b6-5ec032395a75

after:

https://github.com/user-attachments/assets/a8b3f54c-7d66-455e-87ee-1cd42a3ff570

shaik-zeeshan commented:
Can you share the logs?

shaik-zeeshan commented:
> is there a way to display a loading / skeleton like on search when it's loading

Will add this

louis030195 commented:
yes for the image - the "Current frame" and broken logo isn't great haha 


seems both for recent frames:
<img width="1312" alt="Screenshot 2025-02-25 at 9 08 31 AM" src="https://github.com/user-attachments/assets/5cced9c1-e485-4e3c-9692-5ecf7104bd9f" />


and old ones:

<img width="1312" alt="Screenshot 2025-02-25 at 9 09 01 AM" src="https://github.com/user-attachments/assets/9d022482-7f5c-4d7a-a411-9feb47112373" />

shaik-zeeshan commented:
> yes for the image - the "Current frame" and broken logo isn't great haha

yeah. i will loading state

if the file is not created or has invalid data then this is happening 

i am also getting error for some old frames, sometimes
![Screenshot 2025-02-25 at 10 51 44 PM](https://github.com/user-attachments/assets/bf211807-e293-4f56-9672-29be56657730)

when i go to the file and try to open it. it can't be opened

![Screenshot 2025-02-25 at 10 53 44 PM](https://github.com/user-attachments/assets/edc4bff0-40d0-408d-908a-e6e7db08b99d)

shaik-zeeshan commented:
@louis030195 

added  image loading 

https://github.com/user-attachments/assets/8bbe3e9e-9511-45c4-9d56-786767f60293

also fixed frame file not created issue

before
![screenshots](https://github.com/user-attachments/assets/ad78d49f-3e43-4112-b047-07bfcfdbf480)

after
![Screenshot 2025-02-26 at 12 19 12 AM](https://github.com/user-attachments/assets/1a975b1b-cfcb-4b4c-ac99-e179292f518e)

louis030195 commented:
that's awesome 

small issue is that it seems we see more frames now? which makes it hard to use the AI feature over longer ranges 

idea is that could can pinch in/out to zoom on the timeline (currently pinch seems to scroll, shouldn't) that would be rly amazing

shaik-zeeshan commented:
> idea is that could can pinch in/out to zoom on the timeline to see more or less frames (currently pinch seems to scroll, shouldn't) that would be rly amazing

tried to implement this but there issue with scrolling and currentFrame was being reset every time( which won't be good experience). 
currently filtering out the frames before sending it to ai model

louis030195 commented:
so i had to move a few columns from ocr_text to frames if you can adapt the db requests?

https://github.com/mediar-ai/screenpipe/pull/1472

shaik-zeeshan commented:
done

louis030195 commented:
> > idea is that could can pinch in/out to zoom on the timeline to see more or less frames (currently pinch seems to scroll, shouldn't) that would be rly amazing
> 
> tried to implement this but there are issues with scrolling and currentFrame is being reset every time( which won't be good experience). currently filtering out the frames before sending it to ai model

this is not what i meant 

it's just too hard to select like 6 hours of context to chat w AI, that's why a zoom in/out would be great but might be hard to impl 

i'm merging - amazing work overall !

/approve 

few next steps would be:
- zoom in / out 
- more responsive (seems like it's not changing current frame because loading sometimes)
- small stuff (ai chat open, selecting new zone should reset dialog, ..)
- fixing this:
<img width="1483" alt="Screenshot 2025-02-26 at 11 21 56 AM" src="https://github.com/user-attachments/assets/b0e176ef-0505-4078-bdb9-40cecd980533" />

louis030195 commented:
/tip $50 @shaik-zeeshan
